### ISSUE TITLE
Amazon SNS - Bounce not working

### ISSUE THREAD

**Issue by andremacola** [mautic/mautic#7593]
## Bug Description

I had 2.14.x Mautic version and the SNS callback to _mailer/amazon/callback_ was working fine. After upgrading to 2.15.1 I get only 500 status and the bounces are not processed anymore.

Has there been any change?

Mautic version: 2.15.1
PHP version: 7.2.18
Browser: Chrome

### Log errors

```
[2019-06-11 06:07:36] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "You have requested a non-existent service "swiftmailer.mailer.transport.smtp"." at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php line 288 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"swiftmailer.mailer.transport.smtp\". at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php:288)"} []
[2019-06-11 06:15:28] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "You have requested a non-existent service "swiftmailer.mailer.transport.smtp"." at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php line 288 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"swiftmailer.mailer.transport.smtp\". at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php:288)"} []
[2019-06-11 06:15:56] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "You have requested a non-existent service "swiftmailer.mailer.transport.smtp"." at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php line 288 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"swiftmailer.mailer.transport.smtp\". at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php:288)"} []
[2019-06-11 06:16:20] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "You have requested a non-existent service "swiftmailer.mailer.transport.smtp"." at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php line 288 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"swiftmailer.mailer.transport.smtp\". at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php:288)"} []
[2019-06-11 06:16:42] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException: "You have requested a non-existent service "swiftmailer.mailer.transport.smtp"." at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php line 288 {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\ServiceNotFoundException(code: 0): You have requested a non-existent service \"swiftmailer.mailer.transport.smtp\". at /home/mauticoi/public_html/vendor/symfony/dependency-injection/Container.php:288)"} []
```

**Comment by andremacola** [comment]
```
{
    "notification": {
        "messageId": "f3853509-82fc-47fc-873f-4a8b1f290084",
        "topicArn": "arn:aws:sns:us-east-1:116756517457:mautic_bounce",
        "timestamp": "2019-06-11 09:34:27.96"
    },
    "delivery": {
        "deliveryId": "0b10fe45-b9dd-5204-b8df-ab3cc1f5d897",
        "destination": "https://mautic.oimparcial.com.br/mailer/amazon/callback",
        "providerResponse": "Internal Server Error",
        "dwellTimeMs": 68812,
        "attempts": 4,
        "statusCode": 500
    },
    "status": "FAILURE"
}
```

**Comment by andremacola** [comment]
My host was blocking the port 2587 and I was using OTHER OPTIONS. For the callback to work, you need to use Amazon SES but Mautic uses the 2587 port as default in that option and we cannot change in the admin.

So I edit the `AmazonTransport.php` line `71` to `parent::__construct($host, 587, 'tls');`

**Comment by andremacola** [comment]
Ok, I fixed partially, but Mautic is not processing SNS bounce notifications on the callback. I need to use IMAP to check my bounce emails. And I notice Amazon is not delivering all the emails

**Comment by anguschang007** [comment]
I have same issue. Mautic doesn't get any bounced email from Amazon SNS. And getting the following log message while I am visiting the https://myMauticSite.com/mailer/amazon/callback :

`[2019-07-30 09:33:02] mautic.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\FatalThrowableError: "Type error: Argument 1 passed to Mautic\EmailBundle\Swiftmailer\Transport\AmazonTransport::processJsonPayload() must be of the type array, null given, called in /var/www/mautic/app/bundles/EmailBundle/Swiftmailer/Transport/AmazonTransport.php on line 103" at /var/www/mautic/app/bundles/EmailBundle/Swiftmailer/Transport/AmazonTransport.php line 113 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\FatalThrowableError(code: 0): Type error: Argument 1 passed to Mautic\\EmailBundle\\Swiftmailer\\Transport\\AmazonTransport::processJsonPayload() must be of the type array, null given, called in /var/www/mautic/app/bundles/EmailBundle/Swiftmailer/Transport/AmazonTransport.php on line 103 at /var/www/mautic/app/bundles/EmailBundle/Swiftmailer/Transport/AmazonTransport.php:113)"} []`

**Comment by stevedrobinson** [comment]
Same here. I would love to test with an actual http post with a proper json body if I understood what was supposed to be in that json body.

**Comment by npracht** [comment]
Duplicate of https://github.com/mautic/mautic/issues/3655

**Comment by Carlos-mb** [comment]
> I need to use IMAP to check my bounce emails.

Hi 

I'm not able to make Mautic process bounced emails. Not by imap neither SNS. 

SNS reports the same error you explained. 

When I use SMTP, it looks like if it works, the bounce email are read but the contactas are not blocked by mautic. 

I'd thank any help.

Thanks
